File: backend/issue-service/src/main/java/com/example/issueservice/dto/rabbit/IssuePriorityChangedEvent.java
package com.example.issueservice.dto.rabbit;

import com.example.issueservice.dto.models.Issue;

import java.time.Instant;

public record IssuePriorityChangedEvent(
        long projectId,
        long issueId,
        long updaterId,
        String oldPriority,
        String newPriority,
        Instant changedAtUtc
) {
    public static IssuePriorityChangedEvent from(Issue issue, long updaterId, String oldPriority) {
        return new IssuePriorityChangedEvent(
                issue.getProjectId(),
                issue.getId(),
                updaterId,
                oldPriority,
                issue.getPriority().name(),
                Instant.now()
        );
    }
}

File: backend/dashboard-service/Backend.Dashboards.Api/Dto/IssuePriorityChangedEvent.cs
using System.Text.Json.Serialization;

namespace Backend.Shared.DTOs;

public class IssuePriorityChangedEvent
{
    [JsonPropertyName("projectId")]
    public long ProjectId { get; set; }

    [JsonPropertyName("issueId")]
    public long IssueId { get; set; }

    [JsonPropertyName("updaterId")]
    public long UpdaterId { get; set; }

    [JsonPropertyName("oldPriority")]
    public string OldPriority { get; set; } = string.Empty;

    [JsonPropertyName("newPriority")]
    public string NewPriority { get; set; } = string.Empty;

    [JsonPropertyName("changedAtUtc")]
    public DateTime ChangedAtUtc { get; set; } = DateTime.UtcNow;
}